src/prompts/ddl/manager.tsx

```diff
@@ -14,21 +27,47 @@ export enum DdlManagerState {
 export default class DdlManager {
     private readonly context: Context;
     private readonly ddlContainer: RootFrameContainer;
-    private config?: DdlPromptConfig[];
+    private config?: Record<string, DdlPromptConfig>;
+    private activeConfigs?: DdlPromptConfig[] = [];
+    private readonly triggerFilter: PromptTriggerEventFilter<DdlPromptConfig>;
 
     constructor(ctx: Context, rootFrame: RootFrame) {
+        console.info('DdlManager: initialising');
+
         this.ddlContainer = rootFrame.createContainer('ddl');
         this.context = ctx;
 
-        console.info('DdlManager: initialising');
+        this.triggerFilter = new PromptTriggerEventFilter<DdlPromptConfig>(
+            ctx,
+            (_: SdkEvent) => {
+                this.updateActiveConfigs();
+                this.setState(DdlManagerState.READY);
+            }
+        );
 
         this.setState(DdlManagerState.LOADING);
     }
 
-    private onBannerConfirm = async (prompt: DdlPromptConfig) => {
-        this.hidePrompt(prompt);
+    private onBannerConfirm = async (
+        prompt: DdlPromptConfig,
+        components?: FingerprintComponents
+    ) => {
+        if (!!components) {
+            await sendClickRequest(this.context, prompt.uuid, components);
+        }
 
         await deleteDdlBannerConfigFromCache(prompt.uuid);
+
+        this.hidePrompt(prompt);
+
+        const acceptAction = prompt.uiActions.accept;
+
+        const redirectUrl =
```

Comment:
Can we prefer a switch here to exhaust the enum of actions? (prevent extension bugs/overlooking in future)